### ISSUE TITLE
[import/order] Regression in 2.20.1

### ISSUE THREAD

**Issue by South-Paw** [benmosher/eslint-plugin-import#1643]
Installed `2.20.1`, started getting ordering errors like the following;

![image](https://user-images.githubusercontent.com/2154764/73621398-c7402880-469a-11ea-8330-9bf933cf8437.png)

Our rule for `import/order` is

```
'import/order': [
  'error',
  {
    groups: ['builtin', 'external', 'parent', 'sibling', 'index'],
    'newlines-between': 'never',
  },
],
```

Rolling back `eslint-plugin-import` to `2.20.0` resolves the issue.

**Comment by yoyo837** [comment]
Only in Windows.

**Comment by ljharb** [comment]
That makes testing for it difficult; appveyor tests are currently failing for unrelated reasons.

**Comment by targos** [comment]
What I found by debugging:
- The import type of an external module is not determined as such
- The reason is that the `path` passed to `isSubpath` contains Windows separators (backslashes) and the function expects slashes.

**Comment by ljharb** [comment]
`path.normalize` might help us here?

**Comment by kryops** [comment]
This also seems to affect `import/no-extraneous-dependencies`: As the import type is reported as `internal` instead of `external` due to the path containing backslashes, the rule never reports any errors on Windows.

**Comment by kryops** [comment]
I tried to create a minimal reproduction for the `import/order` and `import/no-extraneous-dependencies` rules: https://github.com/kryops/eslint-plugin-import-issue1643

**Comment by ljharb** [comment]
A PR with a failing test case would be most welcome; I can add a fix to that (altho a PR that has both is even better!)

**Comment by targos** [comment]
I think the existing tests should already fail on Windows. I was unable to run the tests on that platform when I tried today, though.

**Comment by raphinesse** [comment]
I had to roll back to `2.19.1` for this issue to go away. `2.20.0` complained about the order of the following imports:
```js
const nearley = require('nearley');
const grammar = require('./grammar');
```

**Comment by ljharb** [comment]
@South-Paw can you check on latest master of this plugin and see if it's fixed?

**Comment by ljharb** [comment]
I'll close this for now, but am happy to reopen if the issue isn't solved on latest master.

**Comment by flymans** [comment]
Still isn't working for me

**Comment by thyde1** [comment]
If this issue has been resolved in master, is there a reason that it has not yet been released?

**Comment by ljharb** [comment]
Because maintainers are humans who aren’t being paid for their time, and can’t always drop everything to cut a release :-)

**Comment by rmlevangelio** [comment]
Hi collaborators, thanks for great plugin. I just want to say that it still happening on Windows.

**Comment by ljharb** [comment]
@rmlevangelio thanks for the report - if you could file a new issue with more details that would be appreciated.
